packages/controller-utils/src/create-service-policy.test.ts

```diff
@@ -2731,76 +3179,379 @@ describe('createServicePolicy', () => {
+                const error = new Error('failure');
+                const mockService = () => {
+                  invocationCounter += 1;
+                  if (invocationCounter === maxRetries + 1) {
+                    return { some: 'data' };
+                  }
+                  throw error;
+                };
+                const policy = createServicePolicy({
+                  maxRetries,
+                  maxConsecutiveFailures,
+                  ...options,
+                });
+
+                const firstExecution = policy.execute(mockService);
+                // It's safe not to await this promise; adding it to the promise
+                // queue is enough to prevent this test from running indefinitely.
+                // eslint-disable-next-line @typescript-eslint/no-floating-promises
+                clock.runAllAsync();
+                await ignoreRejection(firstExecution);
+                clock.tick(duration);
+                const result = await policy.execute(mockService);
+
+                expect(result).toStrictEqual({ some: 'data' });
+              });
+
+              it('calls onAvailable listeners once, even if the service is called more than once', async () => {
+                const maxConsecutiveFailures = 5;
+                const maxRetries = maxConsecutiveFailures;
+                let invocationCounter = 0;
+                const error = new Error('failure');
+                const mockService = () => {
+                  invocationCounter += 1;
+                  if (invocationCounter >= maxRetries + 1) {
+                    return { some: 'data' };
+                  }
+                  throw error;
+                };
+                const onAvailableListener = jest.fn();
+                const policy = createServicePolicy({
+                  maxRetries,
+                  maxConsecutiveFailures,
+                  ...options,
+                });
+                policy.onAvailable(onAvailableListener);
+
+                const firstExecution = policy.execute(mockService);
+                // It's safe not to await this promise; adding it to the promise
+                // queue is enough to prevent this test from running indefinitely.
+                // eslint-disable-next-line @typescript-eslint/no-floating-promises
+                clock.runAllAsync();
+                await ignoreRejection(firstExecution);
+                clock.tick(duration);
+                await policy.execute(mockService);
+                await policy.execute(mockService);
+
+                expect(onAvailableListener).toHaveBeenCalledTimes(1);
+              });
+            });
+          });
+        });
+      });
+    });
+  });
+
+  describe('wrapping a service that succeeds at first and then fails enough to break the circuit', () => {
+    describe.each([
+      {
+        desc: `the default max number of consecutive failures (${DEFAULT_MAX_CONSECUTIVE_FAILURES})`,
```

Comment:
Nit: as mentioned in the other comment, I don't think we need to test the effectiveness of this option in every single test scenario